newdle/client/src/components/MyNewdles.js

```diff
@@ -1,31 +1,12 @@
-import React, {useEffect, useState} from 'react';
+import React from 'react';
 import {Container, Icon, Label, Placeholder} from 'semantic-ui-react';
 import {useHistory} from 'react-router';
 import {serializeDate, toMoment} from '../util/date';
-import styles from './MyNewdles.module.scss';
 import client from '../client';
+import styles from './MyNewdles.module.scss';
 
 export default function MyNewdles() {
-  const [newdles, setNewdles] = useState(null);
-
-  useEffect(() => {
-    let aborted = false;
-    (async () => {
-      let newdles;
-      try {
-        newdles = await client.getMyNewdles();
-      } catch (exc) {
-        return;
-      }
-      if (!aborted) {
-        setNewdles(newdles);
-      }
-    })();
-
-    return () => {
-      aborted = true;
-    };
-  }, []);
+  const [newdles] = client.useBackend(() => client.getMyNewdles(), []);
```

Comment:
@pferreir maybe you have a good idea if this can be made nicer without ending up with react hook dependency hell / having to use useCallback here ;)

Reply:
Instead of passing a closure to `useBackend(...)` you could just pass `client.getMyNewdles` and bind `func` to `this` inside `useBackend`. Unless you think we will ever need to pass a function that shouldn't be bound to `client`. But in that case you could always use a closure like you did here.

Reply:
the problem with that is when you want to pass arguments to the function call.. it'd have to be a separate param of the hook, but then you'd also need to have this in the dependency of the useEffect inside the hook...

Reply:
But you can always just pass a closure if you need to pass arguments too, right?